### DISCUSSION TITLE
Add/delete cell compartments by group

### DISCUSSION THREAD

**Discussion by ewesteinde** [jaxleyverse/jaxley#550] [Q&A] [answered]
Hello, 

Apologies if I missed this in any of the tutorials but is there an easy way to delete compartments/sections of a cell after its been generated? For example, I'm defining a jaxley cell from a .swc file and so automatically generating its groups/branches/compartments using the jx.read_swc function. After creating the cell I then want to delete all compartments associated with the group labeled 'axon' (and actually replace it with a synthetic axon with a defined length & # compartments), which I haven't figured out how to do. Or should I edit the swc file directly before using it to generate a jaxley cell?

Thanks,
Elena

**Comment by michaeldeistler** [comment]
Hi Elena,

Thanks for bringing this up! This feature does not yet exist in `Jaxley`, but I agree that it would be nice to have and we should implement it.

We will keep you updated!
Michael

**Reply by michaeldeistler** [reply]
I also opened an issue [here](https://github.com/jaxleyverse/jaxley/issues/551) to track this.

**Reply by ewesteinde** [reply]
Got it, I'll just modify the swc file then, thanks for the quick reply!

**Reply by michaeldeistler** [reply]
Perfect!

**Comment by jnsbck** [comment, marked as answer]
Hey @ewesteinde,
This is now possible thanks to a flexible interface with `networkX` which we just added in #355. I included an example of pretty much what you were asking in [this tutorial](https://github.com/jaxleyverse/jaxley/blob/main/docs/tutorials/08_importing_morphologies.ipynb). You can either edit the graph before importing it into jaxley with `from_graph`, or edit a Module by exporting it to a graph (`to_graph`) and then re-import it (`from_graph`).
Lemme know in case you have any questions about. :)

**Reply by ewesteinde** [reply]
Perfect, thank you!
